connector-pagerduty/src/main/java/com/redhat/cloud/notifications/connector/pagerduty/PagerDutyTransformer.java

```diff
@@ -0,0 +1,108 @@
+package com.redhat.cloud.notifications.connector.pagerduty;
+
+import io.vertx.core.json.JsonObject;
+import jakarta.enterprise.context.ApplicationScoped;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Constructs a <a href="https://support.pagerduty.com/main/docs/pd-cef">PD-CEF</a> alert event.
+ * <br>
+ * The severity is set to {@link PagerDutySeverity#WARNING}, and the action to {@link PagerDutyEventAction#TRIGGER} for
+ * now. The following optional fields are not set (in jq format): <code>.payload.component, .payload.class, .dedup_key, .links[], .trigger[]</code>
+ * <br>
+ * <ul>
+ *     <li>TODO determine which details to include/remove</li>
+ * </ul>
+ */
+@ApplicationScoped
+public class PagerDutyTransformer implements Processor {
+
+    public static final String PAYLOAD = "payload";
+
+    public static final String ACCOUNT_ID = "account_id";
+    public static final String APPLICATION = "application";
+    public static final String BUNDLE = "bundle";
+    public static final String CLIENT = "client";
+    public static final String CLIENT_URL = "client_url";
+    public static final String CONTEXT = "context";
+    public static final String CUSTOM_DETAILS = "custom_details";
+    public static final String DISPLAY_NAME = "display_name";
+    public static final String ENVIRONMENT_URL = "environment_url";
+    public static final String EVENT_ACTION = "event_action";
+    public static final String EVENT_TYPE = "event_type";
+    public static final String EVENTS = "events";
+    public static final String GROUP = "group";
+    public static final String ORG_ID = "org_id";
+    public static final String SEVERITY = "severity";
+    public static final String SOURCE = "source";
+    public static final String SOURCE_NAMES = "source_names";
+    public static final String SUMMARY = "summary";
+    public static final String TIMESTAMP = "timestamp";
+
+    public static final DateTimeFormatter PD_DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+
+    @Override
+    public void process(Exchange exchange) {
+        JsonObject cloudEventData = exchange.getIn().getBody(JsonObject.class);
+
+        JsonObject cloudEventPayload = cloudEventData.getJsonObject(PAYLOAD);
+        JsonObject message = new JsonObject();
+        message.put(EVENT_ACTION, PagerDutyEventAction.TRIGGER);
+        message.mergeIn(getClientLink(cloudEventPayload, cloudEventData.getString(ENVIRONMENT_URL)));
+
+        JsonObject messagePayload = new JsonObject();
+        messagePayload.put(SUMMARY, cloudEventPayload.getString(EVENT_TYPE));
+        messagePayload.put(TIMESTAMP, LocalDateTime.parse(cloudEventPayload.getString(TIMESTAMP)).format(PD_DATE_TIME_FORMATTER));
+        // TODO is it possible for tenant apps to indicate what severity to use?
```

Comment:
```suggestion
```

This might be a question for PM, so we could leave it like this for now.

Reply:
Yes, let's discuss that in Slack.